### PR TITLE
model: support nvfp4 with axk2

### PR DESCRIPTION
**Pull request by rebel-kblee**
### 🚀 Summary of Changes

support nvfp4 with axk2

> *What does this PR do? What feature, fix, or improvement does it bring?*

---

### 📌 Related Issues / Tickets

* Resolves #
* Related to #

---

### ✅ Type of Change

* [ ] 🚀 Release (`release`)
* [ ] ✨ Feature (`feature`)
* [x] 🧠 Model support (`model`)
* [ ] 🧬 Core engine changes (`core`)
* [ ] 🛠 Bug fix (`fix`)
* [ ] ⚙️ Performance improvement (`perf`)
* [ ] 🔁 Refactor or code cleanup (`refactor`)
* [ ] 📄 Documentation (`docs`)
* [ ] ❓ Other (`other`): please describe

---

### 🧪 How to Test

1. Run `...`
2. Verify output: `...`
3. Edge case tested: `...`

---

### 📸 Screenshots / Logs (if applicable)

---

### 📋 Checklist

* [x] PR title follows Conventional Commits format
* [x] This PR is linked to an existing issue
* [x] The test method is described, and the expected result is clearly stated
* [x] Relevant documentation has been updated (if applicable)

---

### 💬 Notes

---

### CODE REVIEW

File: vllm_rbln/patches/mla.py [hunk 1]
```diff
@@ -30,6 +34,37 @@
...
+@register_patch(
+    target=(
+        "vllm.model_executor.layers.attention.mla_attention."
+        "get_and_maybe_dequant_weights"
+    ),
+    reason=(
+        "The MLA kv_b_proj weight-absorb dequantises eagerly at load time. "
+        "Upstream's fast path covers only Fp8LinearMethod, so per-tensor ModelOpt "
+        "fp8 falls to its generic base case"
+    ),
```
Comment (rebel-cjlee) [inline vllm_rbln/patches/mla.py:54]: The reason string looks like it got cut off. Would you mind adding a sentence about what the base case actually costs us?

File: vllm_rbln/patches/mla.py [hunk 2]
```diff
@@ -30,6 +34,37 @@
...
+    quant_method = getattr(layer, "quant_method", None)
+    if isinstance(quant_method, RBLNModelOptFp8LinearMethod):
+        weight = layer.weight.detach().to("cpu", torch.float32)
+        scale = layer.weight_scale.detach().to("cpu", torch.float32)
+        # ModelOpt fp8 is per-tensor, so the scale is a 0-dim scalar (see
+        # RBLNModelOptFp8LinearMethod.process_weights_after_loading)
+        if scale.ndim != 0:
+            scale = torch.repeat_interleave(scale, torch.tensor(layer.logical_widths))[
+                :, None
+            ]
```
Comment (rebel-cjlee) [inline vllm_rbln/patches/mla.py]: This looks like a dead branch to me. `process_weights_after_loading` should run before the MLA layer's so the scale would already be 0-dim by the time we get here, and the comment right above says the same thing. Am I missing a case where it isn't?

**Review by rebel-cjlee — APPROVED** [review]
